dataplane/linux/policy_mgr.go

```diff
@@ -47,11 +52,35 @@ func newPolicyManager(rawTable, mangleTable, filterTable iptablesTable, ruleRend
 	}
 }
 
+func newRawEgressPolicyManager(rawTable iptablesTable, ruleRenderer policyRenderer, ipVersion uint8, ipSetsCallback func(neededIPSets set.Set)) *policyManager {
+	return &policyManager{
+		rawTable:       rawTable,
+		mangleTable:    &noopTable{},
+		filterTable:    &noopTable{},
+		ruleRenderer:   ruleRenderer,
+		ipVersion:      ipVersion,
+		outboundOnly:   true,
+		ipSetsCallback: ipSetsCallback,
+	}
+}
+
 func (m *policyManager) OnUpdate(msg interface{}) {
 	switch msg := msg.(type) {
 	case *proto.ActivePolicyUpdate:
 		log.WithField("id", msg.Id).Debug("Updating policy chains")
 		chains := m.ruleRenderer.PolicyToIptablesChains(msg.Id, msg.Policy, m.ipVersion)
+		if m.outboundOnly {
+			neededIPSets := set.New()
+			filteredChains := []*iptables.Chain(nil)
+			for _, chain := range chains {
+				if strings.Contains(chain.Name, string(rules.PolicyOutboundPfx)) {
+					filteredChains = append(filteredChains, chain)
+					neededIPSets.AddAll(chain.IPSetIDs())
+				}
+			}
+			chains = filteredChains
+			m.ipSetsCallback(neededIPSets)
```

Comment:
Won't this only cover one policy?

Reply:
Good catch.  Added UT and fixing coding.